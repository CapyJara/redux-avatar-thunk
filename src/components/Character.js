import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  return (
    <>
      <img src={character.image} />
      <h3>{character.name}</h3>
    </>
  );
}

Character.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired
};

export default Character;
