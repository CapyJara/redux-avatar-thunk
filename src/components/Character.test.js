import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a Character', () => {
    const character = {
      image: 'image@image.png',
      name: 'person person'
    };
    const wrapper = shallow(<Character character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
