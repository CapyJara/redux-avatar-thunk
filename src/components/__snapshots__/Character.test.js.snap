// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a Character 1`] = `
<Fragment>
  <img
    src="image@image.png"
  />
  <h3>
    person person
  </h3>
</Fragment>
`;
